go/service/chat_boxer.go

```diff
@@ -169,3 +187,31 @@ func (b *chatBoxer) verify(data []byte, si chat1.SignatureInfo, prefix libkb.Sig
 	_, err := sigInfo.Verify()
 	return (err == nil)
 }
+
+// validSenderKey checks that the key is active for sender at ctime.
+func (b *chatBoxer) validSenderKey(sender gregor1.UID, key []byte, ctime gregor1.Time) (bool, error) {
+	kbSender, err := keybase1.UIDFromString(hex.EncodeToString(sender.Bytes()))
+	if err != nil {
+		return false, err
+	}
+	kid := keybase1.KIDFromSlice(key)
+	t := gregor1.FromTime(ctime)
+
+	user, err := libkb.LoadUser(libkb.NewLoadUserByUIDArg(b.G(), kbSender))
```

Comment:
I think at least this line will be fine for reset.  For deleted user, it will return not found.

Do we want to be able to read messages after the sender is deleted?